Formula/pmd.rb

```diff
@@ -11,11 +11,40 @@ class Pmd < Formula
     sha256 "e5ed61e35b5c980b0e1bcf616bc2615f1c5310ad66ce8f4901ca225764e7e78c" => :mavericks
   end
 
+  depends_on :java => "1.8+"
   depends_on "maven" => :build
 
   def install
     ENV.java_cache
 
+    java_version = ENV["JAVA_HOME"][/((\d\.?)+)\.\d/, 1]
+    toolchains = HOMEBREW_CACHE/"java_cache/.m2/toolchains.xml"
+    rm toolchains if File.exist? toolchains
```

Comment:
@MikeMcQuaid If we're worried about the lack of safefty, one possibility would be NOT to use ENV.java_cache and set a different -Duser.home that's in the buildpath (e.g., buildpath/".brew_home/java_user_home"). Frankly the fact that ENV.java_cache sets a user_home in the cache is quite problematic since it is totally unsafe for multiple brew prefixes to access the same user_home when it doesn't respect formula locking at all.